lib/screens/authentication/recover/recover_account_timer/interactor/usecase/remaining_time_state_mapper.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:hashed/domain-shared/page_state.dart';
+import 'package:hashed/screens/authentication/recover/recover_account_found/interactor/viewmodels/current_remaining_time.dart';
+import 'package:hashed/screens/authentication/recover/recover_account_timer/interactor/viewmodels/recover_account_timer_bloc.dart';
+
+///Seconds in a day
+const int _daySecond = 60 * 60 * 24;
+
+///Seconds in an hour
+const int _hourSecond = 60 * 60;
+
+///Seconds in a minute
+const int _minuteSecond = 60;
+
+class RemainingTimeStateMapper {
+  RecoverAccountTimerState mapResultToState(RecoverAccountTimerState currentState) {
+    int days = 0;
+    int hours = 0;
+    int min = 0;
+    int remainingTimeStamp = currentState.timeRemaining;
+
+    ///Calculate the number of days remaining.
+    if (remainingTimeStamp >= _daySecond) {
+      days = remainingTimeStamp ~/ _daySecond;
+      remainingTimeStamp %= _daySecond;
+    }
+
+    ///Calculate remaining hours.
+    if (remainingTimeStamp >= _hourSecond) {
+      hours = remainingTimeStamp ~/ _hourSecond;
+      remainingTimeStamp %= _hourSecond;
+    } else if (days != 0) {
```

Comment:
not understanding this else if. 

also hours is initialized to 0 anyway, so this statement and else if doesn't change anything. 

why does the number of days - 0 or not - impact the number of hours remaining?